### PR TITLE
Autofill username and password from build config

### PR DESCRIPTION
**Pull request by nbradbury**
This is a feature for developers which enables autofilling the wp.com login so we can avoid entering it every time we install/reinstall/disconnect/reconnect. Simply add these to your `WordPress/gradle.properties`:
```
wp.dotcom_login_username = your_username
wp.dotcom_login_password = your_password
```
If these are non-empty, the signin screen will use them to autofill the username and password.

Note: You may need to rebuild the project for this to work.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 1]
```diff
@@ -228,6 +229,8 @@ public void onClick(View v) {
         initInfoButtons(rootView);
         moveBottomButtons();
 
+        autofillFromBuildConfig();
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: We shoud be cautious and enable it in debug builds only

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 2]
```diff
@@ -330,6 +333,22 @@ public void onClick(View v) {
         mInfoButtonSecondary.setOnClickListener(infoButtonListener);
     }
 
+    /*
+     * autofill the username and password from BuildConfig/gradle.properties (developer feature)
+     */
+    private void autofillFromBuildConfig() {
+        String userName = BuildConfig.DOTCOM_LOGIN_USERNAME;
+        String password = BuildConfig.DOTCOM_LOGIN_PASSWORD;
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: We should do some extra work here to get these by reflection so we can make `wp.dotcom_login_username` and `wp.dotcom_login_password` optional in the property file.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: > We shoud be cautious and enable it in debug builds only

Changed in 113d8aa.

**Comment by nbradbury** [comment]
> We should do some extra work here to get these by reflection so we can make wp.dotcom_login_username and wp.dotcom_login_password optional in the property file.

I'm made this change in 43ea90a but I'm not entirely comfortable with the hack for removing ".beta" from the package name. I'm open to suggestions for a cleaner approach!

**Comment by maxme** [comment]
> I'm made this change in 43ea90a but I'm not entirely comfortable with the hack for removing ".beta" from the package name. I'm open to suggestions for a cleaner approach!

We could do something like:

```diff
-    public static Object getBuildConfigValue(Context context, String fieldName) {
+    public static Object getBuildConfigValue(Application application, String fieldName) {
         try {
             // we must convert org.wordpress.android.beta to org.wordpress.android here
-            String packageName = context.getPackageName().replace(".beta", "");
+            String packageName = application.getClass().getPackage().getName();
             Class<?> clazz = Class.forName(packageName + ".BuildConfig");
             Field field = clazz.getField(fieldName);
             return field.get(null);
```

**Comment by nbradbury** [comment]
> We could do something like:

Good call - that seems to work fine. Change made in b55ca1c.

**Comment by maxme** [comment]
Argh, I forgot [this script](https://github.com/wordpress-mobile/WordPress-Android/blob/c5253cab8f136be04522232def2478078d288393/tools/release-checks.sh#L92-L101) that checks for properties prefixed by `wp.` and compute the sha1 checksum. We need to give them a specific name like `wp.debug` and filter them in the script.

**Comment by maxme** [comment]
> Argh, I forgot this script that checks for properties prefixed by wp. and compute the sha1 checksum. We need to give them a specific name like wp.debug and filter them in the script.

Fixed in 79aa2252330bf6b47c5fba197a8eb7ff38728937

Looks good to me, :shipit: - @nbradbury let me know if 79aa2252330bf6b47c5fba197a8eb7ff38728937 is ok or if you want to add anything.

**Pull request merged**